File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/inferno/ScorchingString.java
package xyz.devvydont.smprpg.items.blueprints.sets.inferno;

import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.RecipeChoice;
import xyz.devvydont.smprpg.items.CustomItemType;
import xyz.devvydont.smprpg.items.ItemClassification;
import xyz.devvydont.smprpg.items.base.CustomItemBlueprint;
import xyz.devvydont.smprpg.items.interfaces.ISellable;
import xyz.devvydont.smprpg.items.interfaces.ISmeltable;
import xyz.devvydont.smprpg.services.ItemService;
import xyz.devvydont.smprpg.util.time.TickTime;

public class ScorchingString extends CustomItemBlueprint implements ISmeltable, ISellable {

    public ScorchingString(ItemService itemService, CustomItemType type) {
        super(itemService, type);
    }

    @Override
    public ItemClassification getItemClassification() {
        return ItemClassification.MATERIAL;
    }

    @Override
    public int getWorth(ItemStack itemStack) {
        return 10_000 * itemStack.getAmount();
    }

    /**
     * Get the ingredient that is used to smelt this item.
     *
     * @return The {@link RecipeChoice} that will turn into this item when cooked.
     */
    @Override
    public RecipeChoice getIngredient() {
        return new RecipeChoice.ExactChoice(ItemService.generate(CustomItemType.ENCHANTED_STRING));
    }

    /**
     * The vanilla Minecraft experience that is awarded as a result for cooking this item.
     *
     * @return The vanilla Minecraft experience.
     */
    @Override
    public float getExperience() {
        return 5;
    }

    /**
     * The cooking time in ticks in order to cook this item.
     *
     * @return The time in ticks.
     */
    @Override
    public long getCookingTime() {
        return TickTime.minutes(5);
    }

    /**
     * Gets the recipe type for this furnace.
     *
     * @return The type of smelting recipe.
     */
    @Override
    public RecipeType getRecipeType() {
        return RecipeType.DEFAULT;
    }
}

File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/inferno/ScorchingString.kt
package xyz.devvydont.smprpg.items.blueprints.sets.inferno

import net.kyori.adventure.key.Key
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.RecipeChoice
import org.bukkit.inventory.RecipeChoice.ExactChoice
import xyz.devvydont.smprpg.items.CustomItemType
import xyz.devvydont.smprpg.items.ItemClassification
import xyz.devvydont.smprpg.items.base.CustomItemBlueprint
import xyz.devvydont.smprpg.items.interfaces.IModelOverridden
import xyz.devvydont.smprpg.items.interfaces.ISellable
import xyz.devvydont.smprpg.items.interfaces.ISmeltable
import xyz.devvydont.smprpg.items.interfaces.ISmeltable.RecipeType
import xyz.devvydont.smprpg.services.ItemService
import xyz.devvydont.smprpg.services.ItemService.Companion.generate
import xyz.devvydont.smprpg.util.time.TickTime

class ScorchingString(itemService: ItemService, type: CustomItemType) : CustomItemBlueprint(itemService, type),
    ISmeltable, ISellable, IModelOverridden {

    override val itemClassification: ItemClassification get() = ItemClassification.MATERIAL

    override fun getWorth(itemStack: ItemStack): Int { return 10000 * itemStack.amount }

    override fun getIngredient(): RecipeChoice { return ExactChoice(generate(CustomItemType.ENCHANTED_STRING)) }

    override fun getExperience(): Float { return 5f }

    override fun getCookingTime(): Long { return TickTime.minutes(5) }

    override fun getRecipeType(): RecipeType { return RecipeType.DEFAULT }

    override fun getDisplayKey(): Key? { return IModelOverridden.ofItemTypeInDirectory(customItemType, "materials") }
}
